common/src/main/java/com/wynntils/screens/partymanagement/PartyManagementScreen.java

```diff
@@ -0,0 +1,426 @@
+                I18n.get("screens.wynntils.partyManagementGui.legend"),
+                this.width / 2 - 300,
+                this.height / 2 - 144,
+                CommonColors.WHITE,
+                HorizontalAlignment.Left,
+                VerticalAlignment.Middle,
+                TextShadow.NORMAL);
+        fr.renderText(
+                poseStack,
+                "§l" + I18n.get("screens.wynntils.partyManagementGui.self"),
+                this.width / 2 - 300,
+                this.height / 2 - 132,
+                CommonColors.WHITE,
+                HorizontalAlignment.Left,
+                VerticalAlignment.Middle,
+                TextShadow.NORMAL);
+        fr.renderText(
+                poseStack,
+                I18n.get("screens.wynntils.partyManagementGui.leader"),
+                this.width / 2 - 300,
+                this.height / 2 - 120,
+                CommonColors.YELLOW,
+                HorizontalAlignment.Left,
+                VerticalAlignment.Middle,
+                TextShadow.NORMAL);
+        fr.renderText(
+                poseStack,
+                "§m" + I18n.get("screens.wynntils.partyManagementGui.offline"),
+                this.width / 2 - 300,
+                this.height / 2 - 108,
+                CommonColors.WHITE,
+                HorizontalAlignment.Left,
+                VerticalAlignment.Middle,
+                TextShadow.NORMAL);
+        fr.renderText(
+                poseStack,
+                I18n.get("screens.wynntils.partyManagementGui.friend"),
+                this.width / 2 - 300,
+                this.height / 2 - 96,
+                CommonColors.GREEN,
+                HorizontalAlignment.Left,
+                VerticalAlignment.Middle,
+                TextShadow.NORMAL);
+        // endregion
+    }
+
+    /**
+     * Reloads the suggested players and their widgets
+     * <p>
+     * This should be called when a friend joins/leaves the world or when the refresh button is pressed
+     */
+    private void reloadSuggestedPlayersWidgets() {
+        // Add friends that are online in the same world as user
+        Scoreboard scoreboard = McUtils.mc().level.getScoreboard();
+        Set<String> onlineUsers = new HashSet<>(scoreboard.getTeamNames());
+        onlineUsers.retainAll(Models.Friends.getFriends());
+
+        List<String> suggestedPlayers = new ArrayList<>(onlineUsers);
+        suggestedPlayers.removeAll(Models.Party.getPartyMembers()); // No need to suggest party members
+        suggestedPlayers.sort(String.CASE_INSENSITIVE_ORDER);
+
+        suggestedPlayersWidgets.clear();
+        for (int i = 0; i < suggestedPlayers.size(); i++) {
+            String playerName = suggestedPlayers.get(i);
+            if (playerName == null) continue;
+
+            suggestedPlayersWidgets.add(new SuggestionPlayer(
+                    this.width / 2 + 204, this.height / 2 - 125 + i * 20 - 10, 200, 20, playerName));
+        }
+    }
+
+    @SubscribeEvent
```

Comment:
Generally we should not use `SubscribeEvent` in screens. (I know I did before, I will remove them sometime).

What I suggest you do: Have the specific models listen to these events, and if the current screen is this, then call methods of this to update it.